lib/classes/PluginManager.js

```diff
@@ -309,6 +310,41 @@ class PluginManager {
     });
   }
 
+  loadVariableResolvers(pluginInstance) {
+    const pluginName = pluginInstance.constructor.name;
+    for (const [variablePrefix, resolverOrOptions] of _.entries(
+      pluginInstance.variableResolvers || {}
+    )) {
+      let options = resolverOrOptions;
+      if (_.isFunction(resolverOrOptions)) {
+        options = {
+          resolver: resolverOrOptions,
+        };
+      }
+      if (!variablePrefix.match || !variablePrefix.match(/[0-9a-zA-Z_-]/)) {
+        throw new Error(
```

Comment:
We should treat it as _user_ error, so we should rather throw `ServerlessError`'s here (then errors are reported clean way without stack trace)

Reply:
Indeed, you're right, plugins while not being part of a core, to user of a framework are still internals, so we shouldn't.

So in that case `TypeError` is the right one